botocore/credentials.py

```diff
@@ -1184,6 +1251,11 @@ def fetch_credentials(require_expiry=True):
                     provider=method, cred_var=mapping['expiry_time']
                 )
 
+            credentials['account_id'] = None
+            account_id = environ.get(mapping['account_id'], '')
+            if account_id:
+                credentials['account_id'] = account_id
+
```

Comment:
This can be written in one line, but I chose to mimic the format used by the other variables above.
`credentials['account_id'] = environ.get(mapping['account_id'])`